orderBook.js

```diff
@@ -1,5 +1,45 @@
-function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+/* eslint-disable indent */
+const reconcileOrder = (existingBook, incomingOrder) => {
```

Comment:
This is pretty complicated for one function, using sub-functions would help readability.